programs/ssl/ssl_pthread_server.c

```diff
@@ -344,26 +400,69 @@ int main( void )
 
     mbedtls_ssl_config_init( &conf );
     mbedtls_ctr_drbg_init( &ctr_drbg );
-    memset( threads, 0, sizeof(threads) );
     mbedtls_net_init( &listen_fd );
     mbedtls_net_init( &client_fd );
 
     mbedtls_mutex_init( &debug_mutex );
 
     base_info.config = &conf;
+    base_info.thread_complete = 0;
 
     /*
      * We use only a single entropy source that is used in all the threads.
      */
     mbedtls_entropy_init( &entropy );
 
+    if( argc == 0 )
```

Comment:
Why `argc == 0`? If `argv[0]` is always the executable filename when called from a shell. `argc` will never equal 0 in any normal use case, and when it does, you don't need to print usage.

Also - can't we run with no parameters, just using the default? Else why have the default? I don't think we need this check.